src/m365/spo/commands/navigation/navigation-node-set.ts

```diff
@@ -0,0 +1,154 @@
+    return commands.NAVIGATION_NODE_SET;
+  }
+
+  public get description(): string {
+    return 'Adds a navigation node to the specified site navigation';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initTypes();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        isExternal: args.options.isExternal,
+        location: typeof args.options.location !== 'undefined',
+        parentNodeId: typeof args.options.parentNodeId !== 'undefined',
+        audienceIds: typeof args.options.audienceIds !== 'undefined'
+      });
+    });
+  }
+
+  #initTypes(): void {
+    this.types.boolean.push('isExternal');
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-u, --webUrl <webUrl>'
+      },
+      {
+        option: '--id <id>'
+      },
+      {
+        option: '--title [title]'
+      },
+      {
+        option: '--url [url]'
+      },
+      {
+        option: '--audienceIds [audienceIds]'
+      },
+      {
+        option: '--isExternal [isExternal]'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        const isValidSharePointUrl: boolean | string = validation.isValidSharePointUrl(args.options.webUrl);
+        if (isValidSharePointUrl !== true) {
+          return isValidSharePointUrl;
+        }
+
+        if (!args.options.audienceIds && !args.options.url && !args.options.isExternal && !args.options.title) {
+          return `Please specify atleast one property to update.`;
+        }
+
+        if (args.options.audienceIds) {
+          const audienceIdsSplitted = args.options.audienceIds.split(',');
+          if (audienceIdsSplitted.length > 10) {
+            return 'The maximum amount of audienceIds per navigation node exceeded. The maximum amount of auciendeIds to be set is 10.';
+          }
+
+          if (!validation.isValidGuidArray(args.options.audienceIds.split(','))) {
+            return `The option audienceIds contains invalid GUIDs`;
+          }
+        }
+
+        return true;
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    if (this.verbose) {
+      logger.logToStderr(`Setting navigation node...`);
+    }
+    logger.log(args.options);
+    const requestBody = this.mapRequestBody(args.options);
+
+    const requestOptions: CliRequestOptions = {
+      url: `${args.options.webUrl}/_api/web/navigation/GetNodeById(${args.options.id})`,
+      headers: {
+        accept: 'application/json;odata=nometadata',
+        'content-type': 'application/json;odata=nometadata'
+      },
+      data: requestBody,
+      responseType: 'json'
+    };
+
+    try {
+      await request.patch(requestOptions);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
```

Comment:
I noticed that this request doesn't fail very often. When I try to break the request, I get following result:

Status code: 200 (Ok)
```json
{
    "odata.null": true
}
```` 

I suggest that we check for this response and throw a clear error for the user that the command failed.